src/components/AdvancedForm.js

```diff
@@ -0,0 +1,82 @@
+import React from "react";
+import {
+  Container,
+  FormControl,
+  FormLabel,
+  FormErrorMessage,
+  FormHelperText,
+  Input,
+  Checkbox,
+  Button,
+} from "@chakra-ui/react";
+
+const AdvancedForm = (props) => {
+  const {
+    handleSubmit,
+    formState,
+    errorsState,
+    handleChange,
+    isButtonDisabled,
+  } = props;
```

Comment:
daca observi pai toate aceste functii tu le declari mai sus si doar le transmiti in jos la AdvancedForm, nu vad de ce ele nu ar putea sta direct in aceasta componenta. Doar nu vrem ca App.js sa aiba grija de logica aditionala cu care nu are a face

Reply:
iar pentru a adauga un user, poti face o functie in App, ceva de genul addUser, pe care o transmiti la AdvancedForm prin props si o invoci dupa ce ai facut post request in onSubmit.

Reply:
Ar avea sens ce zic eu aici? 😄

Reply:
Ok, eu m-am gandit ca asa o  sa fie mai citibil , decat daca as scrie props.handleSubmit etc.. dar o sa modific, merci!

Reply:
stai o secunda, nu e vorba de props.handleSubmit, ci de a migra toate functiile din App.js inclusiv si state-ul care are grija de forma direct in AdvancedForm.js

Reply:
de fapt mie imi place cand props sunt destructurate in felul dat

Reply:
@Ekaterina173 ai timp pentru un quick call? sa explic mai clar poate ce am avut in vedere

Reply:
> iar pentru a adauga un user, poti face o functie in App, ceva de genul addUser, pe care o transmiti la AdvancedForm prin props si o invoci dupa ce ai facut post request in onSubmit.

bine, dar pentru ce sa o transmit in AdvancedForm, daca ea este invocata de Post, dar post se face in App?? nu prea inteleg

Reply:
> stai o secunda, nu e vorba de props.handleSubmit, ci de a migra toate functiile din App.js inclusiv si state-ul care are grija de forma direct in AdvancedForm.js

ok, am inteles ideea

Reply:
deci ideea este de a pune toata logica si toate functiile legate de form direct in AdvancedForm.js, iar tot restul aditional care nu au legatura directa cu componenta sa fie deja prin props

Reply:
> @Ekaterina173 ai timp pentru un quick call? sa explic mai clar poate ce am avut in vedere

Da, nu am vazut acest mesaj

Reply:
bine hai ca te sun pe teams